python/GafferOSLUI/OSLExpressionEngineUI.py

```diff
@@ -0,0 +1,64 @@
+##########################################################################
+#
+#  Copyright (c) 2017, Image Engine Design Inc. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import functools
+
+import Gaffer
+import GafferUI
+
+import OSLUIUtil
+
+def __oslPopupMenu( menuDefinition, widget ) :
+
+	node = widget.node()
+	if not isinstance( node, Gaffer.Expression ) or node.getExpression()[-1] != "OSL" :
+		return
+
+	if len( menuDefinition.items() ) :
+		menuDefinition.append( "/InsertOSLDivider", { "divider" : True } )
+
+	menuDefinition.append(
+		"/Insert OSL",
+		{
+			"subMenu" : functools.partial(
+				OSLUIUtil.commonFunctionMenu,
+				command = widget.textWidget().insertText,
+				activator = lambda : widget.textWidget().getEditable() and not Gaffer.MetadataAlgo.readOnly( node["__expression"] ),
+			),
+		},
+	)
+
+GafferUI.ExpressionUI._ExpressionWidget.popupMenuSignal().connect( __oslPopupMenu, scoped = False )
```

Comment:
Since we've added public methods to _ExpressionWidget, we should also make _it_ public by removing the `_` prefix.